src/main/java/com/exasol/releasedroid/usecases/report/AbstractReport.java

```diff
@@ -0,0 +1,82 @@
+package com.exasol.releasedroid.usecases.report;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.releasedroid.usecases.request.PlatformName;
+
+/**
+ * Abstract part of the report.
+ */
+public abstract class AbstractReport implements Report {
+    private final List<Result> results = new LinkedList<>();
+    private final String reportName;
+    private final PlatformName platformName;
+
+    protected AbstractReport(final String reportName, final PlatformName platformName) {
+        this.reportName = reportName;
+        this.platformName = platformName;
+    }
+
+    @Override
+    public boolean hasFailures() {
+        for (final Result result : this.results) {
+            if (result.isFailed()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public void merge(final Report report) {
+        if (!report.getReportName().equals(getReportName())) {
```

Comment:
this here is breaking the interface contract. I would suggest to rollback this change and have a single Report class